openmdao/recorders/base_recorder.py

```diff
@@ -247,11 +257,38 @@ def startup(self, object_requesting_recording):
                 myresponses = {n for n in object_requesting_recording._responses
                                if self._check_path(n, incl, excl)}
 
+            # get the system_includes that were requested for this Driver recording
+            if sys_incl:
+                prob = object_requesting_recording._problem
+                root = prob.model
+                # The my* variables are sets
+                # sys_incl is not subject to the checking with incl and excl
+                #   sys_incl IS the incl
+
+                # First gather all of the desired outputs
+                # The following might only be the local vars if MPI
+                # mysystem_outputs = {n for n in root._outputs}
+                mysystem_outputs = {n for n in root._outputs
+                                    if self._check_path(n, sys_incl, [])}
+
+                # If MPI, and on rank 0, need to gather up all the variables
+                #    even those not local to rank 0
+                if MPI:
+                    all_vars = root.comm.gather(mysystem_outputs, root=0)
+                    if MPI.COMM_WORLD.rank == 0:
+                        mysystem_outputs = all_vars[-1]
+                        for d in all_vars[:-1]:
+                            mysystem_outputs.update(d)
+                # Filter out only the ones we want
+                # mysystem_outputs = {n for n in mysystem_outputs
```

Comment:
Did we want the filtering commented out, or is this done elsewhere now?

Reply:
fixed and pushed up